src/game/boe.main.cpp

```diff
@@ -245,21 +319,38 @@ void init_boe(int argc, char* argv[]) {
 #ifdef __APPLE__
 	init_menubar(); // Do this first of all because otherwise a default File and Window menu will be seen
 #endif
+
+	// TODO preferences need to be recorded in the action log for deterministic replay
 	sync_prefs();
 	init_shaders();
 	init_tiling();
 	init_snd_tool();
 	
 	init_ui();
-	
+
 	adjust_window_mode();
 	// If we don't do this now it'll flash white to start with
 	mainPtr.clear(sf::Color::Black);
 	mainPtr.display();
 	
 	set_cursor(watch_curs);
 	init_buf();
-	srand(time(nullptr));
+
+	// Seed the RNG
+	if (replaying) {
+		auto srand_element = pop_next_action("srand");
+		
+		std::string ts(srand_element->GetText());
+		srand(atoi(ts.c_str()));
```

Comment:
Seeing this made me realize that we may want to consider moving away from `rand()` in favour of a Mersenne twister… but that would be something for a separate PR.